tasks/eva_2877/load_to_contig_alias_db.py

```diff
@@ -0,0 +1,100 @@
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query
+from retry import retry
+
+logging_config.add_stdout_handler()
+logger = logging_config.get_logger(__name__)
+
+
+def get_assemblies_from_evapro(private_config_xml_file):
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        query = "select distinct assembly_accession from accessioned_assembly where assembly_accession like 'GCA%'"
+        evapro_assemblies = get_all_results_for_query(pg_conn, query)
+        return [asm[0] for asm in evapro_assemblies]
+
+
+def get_contig_alias_auth(private_config_xml_file):
+    properties = get_properties_from_xml_file("development", private_config_xml_file)
+    contig_alias_username = str(properties['contig-alias.admin-user'])
+    contig_alias_password = str(properties['contig-alias.admin-password'])
+
+    auth = f"{contig_alias_username}:{contig_alias_password}"
+    encode_auth = auth.encode('ascii')
+    base64_auth = base64.b64encode(encode_auth)
+    return base64_auth.decode()
+
+
+def load_assembly_to_contig_alias(assemblies, admin_credentials, delete_insert):
+    for assembly in assemblies:
+        url = f"https://wwwdev.ebi.ac.uk/eva/webservices/contig-alias/v1/admin/assemblies/{assembly}"
+        headers = {'Authorization': 'Basic ' + admin_credentials}
+        if delete_insert == 'True':
+            # delete request for assembly
+            del_request(assembly, url, headers)
+
+        # insert request for assembly
+        insert_request(assembly, url, headers)
+
+
+def del_request(assembly, url, headers):
+    try:
+        create_and_execute_request(url, 'DELETE', headers)
+        logger.info(f'Assembly info Deleted successfully for Assembly {assembly}')
+    except Exception as e:
+        logger.error(f'Error Deleting assembly {assembly}. Exception : {e}')
+
+
+def insert_request(assembly, url, headers):
+    try:
+        create_and_execute_request(url, 'PUT', headers)
+        logger.info(f'Assembly info inserted successfully for Assembly {assembly}')
+    except HTTPError as error:
+        logger.error(f'Error inserting assembly {assembly}. HTTP Error: {error}')
+    except URLError as error:
+        if isinstance(error.reason, socket.timeout):
+            logger.error(f'Error inserting assembly {assembly}. URL Timeout Error: {error}')
+        else:
+            logger.error(f'Error inserting assembly {assembly}. URL Error: {error}')
+    except Exception as e:
+        logger.error(f'Error inserting assembly {assembly}. Exception : {e}')
+
+
+class CustomError(Exception):
+    pass
+
+@retry(CustomError, tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def create_and_execute_request(url, method, headers):
+    client_request = urllib.request.Request(url, None, headers, method=method)
```

Comment:
Using `requests` library is a bit cleaner especially for the auth header, you can look at the code in [submission automation](https://github.com/EBIvariation/eva-submission/blob/0d4abeed9d7dbde1c1f4b2b9e264b26ab23c04b9/eva_submission/eload_preparation.py#L199).

Reply:
done